core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -807,8 +801,8 @@ private[spark] class BlockManager(
         // Now that the block is in either the memory or disk store,
         // tell the master about it.
         info.size = size
-        if (tellMaster) {
-          reportBlockStatus(blockId, info, putBlockStatus)
+        if (tellMaster && info.tellMaster) {
```

Comment:
Yep, looks like my memory was correct:

Take a look at [ShuffleMapTask in Spark v0.6.0](https://github.com/apache/spark/blame/v0.6.0/core/src/main/scala/spark/scheduler/ShuffleMapTask.scala#L151):

```
val blockManager = SparkEnv.get.blockManager
    for (i <- 0 until numOutputSplits) {
      val blockId = "shuffle_" + dep.shuffleId + "_" + partition + "_" + i
      // Get a Scala iterator from Java map
      val iter: Iterator[(Any, Any)] = bucketIterators(i)
      val size = blockManager.put(blockId, iter, StorageLevel.DISK_ONLY, false)
      compressedSizes(i) = MapOutputTracker.compressSize(size)
    }
```

Here, `false` is the `tellMaster` parameter.

This functionality in the BlockManager dates back to the original Spark Streaming engine improvement patch: https://github.com/apache/spark/commit/63051dd2bcc4bf09d413ff7cf89a37967edc33ba